frontend/src/views/DashboardPage/components/SecretInputRow/SecretInputRow.tsx

```diff
@@ -218,6 +225,15 @@ export const SecretInputRow = memo(
       return <></>;
     }
 
+    let isPasted = false; 
+    
+    const handlePaste = () => {
```

Comment:
Using `onPaste` event for this is indeed a good move. 

But using this way makes it so complicated. I would request you to revert back the onchange things and instead here you will get the event and then using that event u will get clipboard content. Using `setValue` function from react hook form and setEditor ref set the things accordingly